README.md

```diff
@@ -16,22 +16,40 @@ any zoho api you need.
 composer require weble/zohoclient 
 ```
 
-## Example Usage (Offline Mode)
+## Example Usages (Offline Mode)
+
+### Retrieve the url to authenticate against ZOHO and retrieve the access Token / Refresh Token for the first time
 ```php
 require_once './vendor/autoload.php';
 
-$client = new \Weble\ZohoClient\OAuthClient('{CLIENT_ID}', '{CLIENT_SECRET}');
-$client->setRefreshToken('{REFRESH_TOKEN}');
-$client->setRegion(\Weble\ZohoClient\Enums\Region::us());
-$client->offlineMode();
-$client->useCache($yourPSR6CachePool);
+$client = new \Weble\ZohoClient\OAuthClient('{CLIENT_ID}', '{CLIENT_SECRET}', '{REGION}', '{REDIRECTURL}');
+$client->offlineMode(); // this needs to be set if you want to be able to refresh the token
 
-// Done!
-$accessToken = $client->getAccessToken();
+// Get the url
+$client->setScopes([]); // Set the zoho scopes you need, see https://www.zoho.com/crm/developer/docs/api/v2/scopes.html
+$url = $client->getAuthorizationUrl();
+$state = $client->getState(); // Get the state for security, and save it (usually in session)
 
-// Check if it's expired
-$isExpired = $client->accessTokenExpired();
+redirect($url); // Do your redirection as you prefer
+
+// Wait for the user to redirect...
 
+// In the redirection page, check for the state you got before and that you should've stored
+if ($state !== $_GET['state']) {
```

Comment:
This is under the section "offline mode" which would mean $_GET would not be available

Reply:
This is a generic example on how to get a refresh token with this library. I agree i should not put it under offline mode lol

Reply:
That's a really confusing code example. It looks like one block of code but is actually various different things. What is the redirect() method? Is this from some framework like Laravel? Also, if there has been a redirect then the code below that will never even run.

In offline mode the code example should be split into the different processes: 

1) setting up the initial authorisation with the OAuth server including saving the state to session for checking later
2) a callback that picks up the redirect from the OAuth server, checks the state for security, then saves the grant token and access token to whatever storage provider you wish
3) an example showing how to use the saved grant token to make an API call to an authorised resource

Reply:
@bluec `redirect()` is just a placeholder. Fill it in with whatever you would use to redirect

There is no redirect during offline mode, nor is there a grant token (Perhaps you mean online mode)

Reply:
@tm1000 it doesn't make sense to use functions that don't exist in a code example. The purpose of an example is to ease confusion, not add confusion. The example should use:

```
header('Location: ' . $client->getAuthorizationUrl());
exit;
```

It is not true to say there is no redirect during offline mode. There is a redirect during the initial authorisation flow: it occurs when you pass the user off to the Oauth2 server for authorisation.

You're right about grant token, I meant access token and refresh token.

Reply:
@bluec that was just a stub for explaining how to get a refresh token (zoho is confusing about that and a lot of people ask that to me).

Maybe you can propose a better way with a PR? :)